examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart

```diff
@@ -0,0 +1,202 @@
+    return ReminderListResponse(items: items);
+  }
+
+  @override
+  Future<int> getCompleteCount() async => (await _remindersBox)
+      .query(ObjectBoxLocalReminderModel_.complete.equals(true))
+      .build()
+      .find()
+      .length;
+
+  @override
+  Future<int> getIncompleteCount() async => (await _remindersBox)
+      .query(ObjectBoxLocalReminderModel_.complete.equals(false))
+      .build()
+      .find()
+      .length;
+
+  @override
+  Future<ReminderPair> update(ReminderModel updatedModel) async {
+    var oldModel = (await _remindersBox).get(int.parse(updatedModel.id));
+    if (oldModel == null) {
+      throw Exception('Old model not found');
+    }
+    final oldModelId = int.tryParse(updatedModel.id);
+    if (oldModelId == null) {
+      throw Exception('Old model id is not a number');
+    }
+    (await _remindersBox).put(updatedModel.toReminderObjectboxLocalModel(
+      oldModelId,
+      updatedModel.title,
+      updatedModel.dueDate,
+      updatedModel.complete,
+      updatedModel.authorId,
+    ));
+
+    return ReminderPair(
+      old: oldModel.toReminderModel(),
+      updated: updatedModel,
+    );
+  }
+
+  /// Returns the id of the logged in user or null if no user is logged in
+  Future<String?> _getAuthorIdOrNull() async {
+    var user = await _storage.read(key: _authorId);
+    if (user == _anonymous) {
+      return null;
+    }
+    return user;
+  }
+
+  Condition<ObjectBoxLocalReminderModel> getCondition(
+      ReminderModelRequest? request, String? authorId) {
+    Condition<ObjectBoxLocalReminderModel>? condition;
+    DateTime from;
+    DateTime to;
+
+    if (request?.filterByDueDateRange != null) {
+      from = request?.filterByDueDateRange?.from ?? DateTime.now();
+      to = request?.filterByDueDateRange?.to ?? DateTime.now();
+    } else {
+      from = DateTime.now().subtract(const Duration(days: 365));
+      to = DateTime.now().add(const Duration(days: 365));
+    }
+    condition = ((authorId == null
+        ? ObjectBoxLocalReminderModel_.authorId.isNull()
+        : ObjectBoxLocalReminderModel_.authorId.equals(authorId)));
+    if (request?.complete != null) {
+      condition.and(
+          ObjectBoxLocalReminderModel_.complete.equals(request!.complete!));
+    }
+
+    condition.and(ObjectBoxLocalReminderModel_.dueDate.betweenDate(from, to));
+    return condition;
+  }
+
+  int getOrder(ReminderModelRequestSort? sort) {
+    int order = 0;
+    if (sort != null) {
+      switch (sort) {
+        case ReminderModelRequestSort.dueDateAsc:
+          break;
+        case ReminderModelRequestSort.dueDateDesc:
+          order = Order.descending;
+      }
+    }
+    return order;
+  }
+}
+
+extension ReminderModelMapper on ReminderModel {
+  ObjectBoxLocalReminderModel toReminderObjectboxLocalModel(
+      int id, String title, DateTime dueDate, bool complete, String? authorId) {
+    return ObjectBoxLocalReminderModel(
```

Comment:
Considering none of the ReminderModel's property is not used what's the point of creating an extension on top of that model ?

Reply:
I'm not sure that I understand this

Reply:
![image](https://github.com/Prime-Holding/rx_bloc/assets/32831935/300e45aa-6fd1-46f9-a77c-bd43b2a02de0)

Make sure you follow the established project structure